packages/app/features/Receive/ReceiveModal.tsx

```diff
@@ -67,15 +79,60 @@ const ReceiveModal: FC<{ config: ModalConfigs }> = ({ config }) => {
 		context: { cardList: walletList },
 	};
 
+	const handleClose = () => {
+		modalActions.hide(config?.id as string);
+	};
+
+	const offset = useSharedValue(0);
+	const context = useSharedValue(0);
+	const modalHeight = useSharedValue(0);
+
+	const pan = Gesture.Pan()
+		.onStart(() => {
+			context.value = offset.value;
+		})
+		.onUpdate((event) => {
+			const newPosition = event.translationY + context.value;
+			if (newPosition > 0) {
+				offset.value = newPosition;
+			}
+		})
+		.onEnd((event) => {
+			const newPosition = event.translationY + context.value;
+			if (newPosition > 100) {
+				offset.value = withTiming(modalHeight.value);
+				runOnJS(handleClose)();
+			} else {
+				offset.value = withTiming(0);
+			}
+		});
```

Comment:
I would prefer separate onStart, onUpdate, onEnd..  in to methods that would be more generic and re-useable:
```typescript
const onGestureStart = () => {
  context.value = offset.value;
};

const onGestureUpdate = () => { .... };

const panGesture = Gesture.pan()
  .onStart(onGestureStart)
  .onUpdate(onGestureUpdate)
  .onEnd(onGestureEnd);
```